homeassistant/components/notify/pushbullet.py

```diff
@@ -123,27 +124,27 @@ def send_message(self, message=None, **kwargs):
             # Attempt push_note on a dict value. Keys are types & target
             # name. Dict pbtargets has all *actual* targets.
             try:
-                if url:
-                    self.pbtargets[ttype][tname].push_link(
-                        title, url, body=message)
-                else:
-                    self.pbtargets[ttype][tname].push_note(title, message)
+                self._push_data(filepath, message, title, url,
+                                self.pbtargets[ttype][tname])
                 _LOGGER.info("Sent notification to %s/%s", ttype, tname)
             except KeyError:
                 _LOGGER.error("No such target: %s/%s", ttype, tname)
                 continue
-            except self.pushbullet.errors.PushError:
-                _LOGGER.error("Notify failed to: %s/%s", ttype, tname)
-                continue
 
-    def _push_data(self, filepath, message, title, url, tname=None):
-        if url:
-            self.pushbullet.push_link(
-                title, url, body=message, email=tname)
-        elif filepath and self.hass.config.is_allowed_path(filepath):
-            with open(filepath, "rb") as fileh:
-                filedata = self.pushbullet.upload_file(fileh, filepath)
-                self.pushbullet.push_file(title=title, body=message,
-                                          **filedata)
-        else:
-            self.pushbullet.push_note(title, message, email=tname)
+    def _push_data(self, filepath, message, title, url, pusher, tname=None):
```

Comment:
Why do you need to pass in `pusher` instead of referring to `self.pushbullet` ?

Reply:
When sending to a specific target `pusher` need to be `self.pbtargets[ttype][tname]`
https://github.com/home-assistant/home-assistant/pull/9128/files#diff-087ff967696110e533ac2bb2edf3f344R128